forge/src/gas_report.rs

```diff
@@ -0,0 +1,147 @@
+use crate::{
+    executor::{CHEATCODE_ADDRESS, HARDHAT_CONSOLE_ADDRESS},
+    trace::{CallTraceArena, RawOrDecodedCall, TraceKind},
+};
+use comfy_table::{modifiers::UTF8_ROUND_CORNERS, presets::UTF8_FULL, *};
+use ethers::types::U256;
+use serde::{Deserialize, Serialize};
+use std::{collections::BTreeMap, fmt::Display};
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+pub struct GasReport {
+    pub report_for: Vec<String>,
+    pub contracts: BTreeMap<String, ContractInfo>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct ContractInfo {
+    pub gas: U256,
+    pub size: U256,
+    pub functions: BTreeMap<String, GasInfo>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct GasInfo {
+    pub calls: Vec<U256>,
+    pub min: U256,
+    pub mean: U256,
+    pub median: U256,
+    pub max: U256,
+}
+
+impl GasReport {
```

Comment:
This entire struct is pretty much the same except for a few changes:

- `finalize` now takes `mut self` and returns `Self` so we can do `println!("{}", report.finalize())` - I couldn't see why we would keep the report around after that point anyway
- Pulling function names, contract names etc. has been adjusted to the new call trace identify/decode loop which changed the data structures a bit